lib/media/streaming_engine.js

```diff
@@ -1267,7 +1267,21 @@ shaka.media.StreamingEngine = class {
             'ReadableStream is not supported by the browser.');
         }
         const fetchSegment = this.fetch_(mediaState, reference);
-        const result = await fetchSegment;
+        let result = await fetchSegment;
+        if (stream.hlsAes128Key) {
+          let iv = stream.hlsAes128Key.iv;
+          if (!iv) {
+            iv = shaka.util.BufferUtils.toUint8(new ArrayBuffer(16));
+            let sequence = stream.hlsAes128Key.firstMediaSequenceNumber +
```

Comment:
Does your test content have explicit IVs in the playlist?  If so, have you tested with some other content that exercises this code?

Could you please make sure we have examples of both in the demo asset list?

Reply:
Yes, test contents are listed in the merge message. I don't know about the video license so I only added the "Const IV" one to the demo.

https://playertest.longtailvideo.com/adaptive/oceans_aes/oceans_aes.m3u8 (No IV)
https://bitmovin-a.akamaihd.net/content/art-of-motion_drm/m3u8s/11331.m3u8 (Const IV)